tools/src/main/java/org/apache/kafka/trogdor/workload/ExternalCommandWorker.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.kafka.trogdor.common.WorkerUtils;
+import org.apache.kafka.trogdor.task.TaskWorker;
+import org.apache.kafka.trogdor.task.WorkerStatusTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+
+import java.util.ArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * ExternalCommandWorker drives the Trogdor task with an external command.
+ *
+ * ExternalCommandWorker starts an external process to execute the command. Communication between the external process
+ * and ExternalCommandWorker should follow these APIs:
+ *
+ * Control API:
+ *
+ * ExternalCommandWorker should pass one argument, --spec taskSpec, that shows the Trogdor task.
+ *
+ * ExternalCommandWorker can send a stop command to the external process's stdin, formatted as JSON object {"action":"stop"}.
+ *
+ * Communication API:
+ *
+ * ExternalCommandWorker should continuously monitor the stdout and the stderr of the external process line by line.
+ *
+ * For any JSON object the external process writes to its stdout, if the object has the "status" field, this worker should set
+ * its status with the "status" value, if the object has the "error" field and the value is a JSON string, this worker should
+ * set the agent error message with the "error" value, and if the object has the "log" field, this worker should put the "log"
+ * value it its log.
+ *
+ * For any line the external process writes to its stderr, this worker should save the line to its error log and set the
+ * error message with the line.
+ */
+public class ExternalCommandWorker implements TaskWorker {
+    private static final Logger log = LoggerFactory.getLogger(ProduceBenchWorker.class);
+
+    private final String id;
+
+    private final ExternalCommandSpec spec;
+
+    private final AtomicBoolean running = new AtomicBoolean(false);
+
+    private ScheduledExecutorService executor;
+
+    private Process worker;
+
+    private String errMsg = "";
+
+    private WorkerStatusTracker status;
+
+    private KafkaFutureImpl<String> doneFuture;
+
+    public ExternalCommandWorker(String id, ExternalCommandSpec spec) {
+        this.id = id;
+        this.spec = spec;
+    }
+
+    @Override
+    public void start(Platform platform, WorkerStatusTracker status,
+                      KafkaFutureImpl<String> doneFuture) throws Exception {
+        if (!running.compareAndSet(false, true)) {
+            throw new IllegalStateException("ExternalCommandWorker is already running.");
+        }
+        log.info("{}: Activating ExternalCommandWorker with {}", id, spec);
+        this.executor = Executors.newScheduledThreadPool(2,
```

Comment:
I don't think limiting the pool to 2 threads is really what you want, since you have 3 runnables that you are creating later on.  Check out some of the other newXYZThreadPool functions.  There is at least one that doesn't limit the number of threads.